Sources/BraveWallet/Crypto/Stores/KeyringStore.swift

```diff
@@ -439,13 +456,8 @@ extension KeyringStore: BraveWalletKeyringServiceObserver {
 
 extension KeyringStore: BraveWalletJsonRpcServiceObserver {
   public func chainChangedEvent(_ chainId: String, coin: BraveWallet.CoinType, origin: URLOrigin?) {
-    Task { @MainActor in // This observer will take care of selected account update caused by network switching
-      let accountAddress = await keyringService.selectedAccount(coin)
-      let keyring = await keyringService.keyringInfo(coin.keyringId)
-      if let account = keyring.accountInfos.first(where: { $0.address == accountAddress }) {
-        selectedAccount = account
-      }
-    }
+    walletService.setSelectedCoin(coin)
```

Comment:
feel like we've been discussing this before. but i still think selected coin should be automatically updated when network or account is changed in core.

Reply:
I think it's because they're different services in core. Soon I believe this global selected coin would only be used in the wallet panel, once we can add unapproved transactions for the non-selected chain: https://github.com/brave/brave-browser/issues/30287